ios/RNTPTDocumentView.m

```diff
@@ -5923,3 +5933,33 @@ - (BOOL)isAnnotationStateEnabled
 
 @end
 
+#pragma mark - RNTPTDigitalSignatureTool
+
+@implementation RNTPTDigitalSignatureTool
+
+- (void)signaturesManagerNumberOfSignaturesDidChange:(PTSignaturesManager *)signaturesManager numberOfSignatures:(int)numberOfSignatures
+{
+    [super signaturesManagerNumberOfSignaturesDidChange:signaturesManager numberOfSignatures:numberOfSignatures];
+    
+    if ([self.toolManager.viewController isKindOfClass:[RNTPTDocumentViewController class]]) {
+        RNTPTDocumentViewController *viewController = (RNTPTDocumentViewController *) self.toolManager.viewController;
+        
+        if ([viewController.delegate respondsToSelector:@selector(rnt_documentViewControllerSavedSignaturesChanged:)]) {
+            [viewController.delegate rnt_documentViewControllerSavedSignaturesChanged:viewController];
+        }
+    } else if ([self.toolManager.viewController isKindOfClass:[RNTPTDocumentController class]]) {
+        RNTPTDocumentController *viewController = (RNTPTDocumentController *) self.toolManager.viewController;
+        
+        if ([viewController.delegate respondsToSelector:@selector(rnt_documentViewControllerSavedSignaturesChanged:)]) {
+            [viewController.delegate rnt_documentViewControllerSavedSignaturesChanged:viewController];
+        }
+    } else if ([self.toolManager.viewController isKindOfClass:[RNTPTCollaborationDocumentController class]]) {
+        RNTPTCollaborationDocumentController *viewController = (RNTPTCollaborationDocumentController *) self.toolManager.viewController;
+        
+        if ([viewController.delegate respondsToSelector:@selector(rnt_documentViewControllerSavedSignaturesChanged:)]) {
+            [viewController.delegate rnt_documentViewControllerSavedSignaturesChanged:viewController];
+        }
+    }
```

Comment:
This last else isn't necessary because `RNTPTCollaborationDocumentController` is a subclass of `RNTPTDocumentController`, and it will be covered by that section. (`isMemberOfClass:` returns true if the class types are equal; `isKindOfClass` returns true if the class is equal to or a subclass of the other class.)

Reply:
Actually hang on a sec, I didn't realize these were both the RN types. In the case you can cover both by checking if they are `isKindOfClass` `PTDocumentController`

Reply:
You can ignore this comment original comment, I'll make a fresh one.

Reply:
Actually looking at the class structure, it does have to be done this way.